modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java

```diff
@@ -0,0 +1,442 @@
+            String title = openAPI.getInfo().getTitle();
+
+            // Drop any API suffix
+            if (title != null) {
+                title = title.trim().replace(" ", "-");
+                if (title.toUpperCase().endsWith("API"))
+                    title = title.substring(0, title.length() - 3);
+
+                this.title = camelize(sanitizeName(title), true);
+            }
+            additionalProperties.put(TITLE, this.title);
+        }
+
+        if (!additionalProperties.containsKey(SERVER_PORT)) {
+            URL url = URLPathUtils.getServerURL(openAPI);
+            this.additionalProperties.put(SERVER_PORT, URLPathUtils.getPort(url, 8080));
+        }
+
+        // TODO: Handle tags
+    }
+
+    @Override
+    public void postProcessModelProperty(CodegenModel model, CodegenProperty property) {
+        super.postProcessModelProperty(model, property);
+
+        if ("null".equals(property.example))
+            property.example = null;
+
+        //Add imports for Jackson
+        if (!Boolean.TRUE.equals(model.isEnum)) {
+            model.imports.add("JsonProperty");
+            if (Boolean.TRUE.equals(model.hasEnums))
+                model.imports.add("JsonValue");
+
+        } else {
+            //Needed imports for Jackson's JsonCreator
+            if (additionalProperties.containsKey("jackson"))
+                model.imports.add("JsonCreator");
+        }
+
+    }
+
+    @Override
+    public Map<String, Object> postProcessModelsEnum(Map<String, Object> objs) {
+        objs = super.postProcessModelsEnum(objs);
+
+        //Add imports for Jackson
+        List<Map<String, String>> imports = (List<Map<String, String>>) objs.get("imports");
+        List<Object> models = (List<Object>) objs.get("models");
+
+        models.stream()
+                .map(mo -> (Map<String, Object>) mo)
+                .map(mo -> (CodegenModel) mo.get("model"))
+                .filter(cm -> Boolean.TRUE.equals(cm.isEnum) && cm.allowableValues != null)
+                .forEach(cm -> {
+                    cm.imports.add(importMapping.get("JsonValue"));
+                    Map<String, String> item = new HashMap<>();
+                    item.put("import", importMapping.get("JsonValue"));
+                    imports.add(item);
+                });
+
+        return objs;
+    }
+
+    @Override
+    public Map<String, Object> postProcessOperationsWithModels(Map<String, Object> objs, List<Object> allModels) {
+        Map<String, Object> operations = (Map<String, Object>) objs.get("operations");
+        if (operations != null) {
+            List<CodegenOperation> ops = (List<CodegenOperation>) operations.get("operation");
+            ops.forEach(operation -> {
+                List<CodegenResponse> responses = operation.responses;
+                if (responses != null) {
+                    responses.forEach(resp -> {
+
+                        if ("0".equals(resp.code))
```

Comment:
I love that this setting a "good" response code if none was defined in the spec. 

And it makes me wonder if we should loop responses and verify that response codes are valid as part of the `cli validate --recommend` option.

@jmini @ackintosh any thoughts on adding that as a feature (because you both worked with me most on the feature and requirements)?